### PR TITLE
Use layer types and versions from MapLayerService in Wizard

### PR DESCRIPTION
**Pull request by mikkomustikkamaa**
This PR contains changes to use layer types and versions from MapLayerService in layer wizard. WFS and WMS layer plugins provide their own supported versions to MapLayerService in registerLayerModel call.

### CODE REVIEW

File: bundles/mapping/mapmodule/service/map.layer.js [hunk 1]
```diff
@@ -939,9 +941,27 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
...
          *            layer model clazz name (like 'Oskari.mapframework.domain.WmsLayer') or constructor function
+         * @param {Array} availableVersionsForType
+         *            string array containing available versions for given type
          */
-        registerLayerModel: function (type, modelRef) {
+        registerLayerModel: function (type, modelRef, availableVersionsForType) {
             this.typeMapping[type] = modelRef;
+            if (availableVersionsForType) {
+                this.availableVersions[type] = availableVersionsForType;
+                this.notify();
```
Comment (jheijari) [inline bundles/mapping/mapmodule/service/map.layer.js]: Use observable's `trigger` instead.

File: bundles/mapping/mapmodule/service/map.layer.js [hunk 2]
```diff
@@ -939,9 +941,27 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
...
+            return this.availableVersions[type] || [];
+        },
+        getLayerTypes () {
+            return Object.keys(this.availableVersions) || [];
+        },
+        addStateListener (consumer) {
+            this.stateListeners.push(consumer);
+        },
+        notify () {
+            this.stateListeners.forEach(consumer => consumer());
```
Comment (jheijari) [inline bundles/mapping/mapmodule/service/map.layer.js]: Replace logic with observable's event functions.

File: bundles/mapping/mapmodule/service/map.layer.js [hunk 3]
```diff
@@ -75,6 +76,7 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
         };
 
         Oskari.makeObservable(this);
+        this.stateListeners = [];
```
Comment (jheijari) [inline bundles/mapping/mapmodule/service/map.layer.js]: Remove `stateListeners`, the service is already observable.

File: bundles/admin/admin-layereditor/view/LayerWizard.jsx [hunk 4]
```diff
@@ -58,7 +58,8 @@ const LayerWizard = ({
     layerTypes = [],
     loading,
     children,
-    Message
+    Message,
```
Comment (jheijari) [inline bundles/admin/admin-layereditor/view/LayerWizard.jsx]: Remove Message prop.

File: bundles/admin/admin-layereditor/view/Flyout.js [hunk 5]
```diff
@@ -16,6 +16,8 @@ export class LayerEditorFlyout extends ExtraFlyout {
         this.dataProviders = [];
         this.mapLayerGroups = [];
         this.service = new AdminLayerFormService(() => this.update());
+        this.mapLayerService = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
+        this.mapLayerService.addStateListener(() => this.update());
```
Comment (jheijari) [inline bundles/admin/admin-layereditor/view/Flyout.js]: Use as observable. `this.mapLayerService.on('some-event-name', () => this.update())`

File: bundles/mapping/mapmodule/service/map.layer.js (hunk 2 above)
Comment (mikkomustikkamaa) [reply bundles/mapping/mapmodule/service/map.layer.js]: Done

File: bundles/mapping/mapmodule/service/map.layer.js (hunk 3 above)
Comment (mikkomustikkamaa) [reply bundles/mapping/mapmodule/service/map.layer.js]: Done

File: bundles/admin/admin-layereditor/view/Flyout.js (hunk 5 above)
Comment (mikkomustikkamaa) [reply bundles/admin/admin-layereditor/view/Flyout.js]: I tried to follow similar notification mechanism that is used with AdminLayerFormService in same flyout and for this reason implementation was done as it was. Implementation has now been changed as suggested in review comments.

**Pull request merged**